Do not copy bin subdirectories to test core_root

mikedn opened the pull request:
As far as I can tell there's no need to copy various bin subdirectories (PDB, .nuget, IL etc.) to core_root when running tests and they have hundreds of megabytes.

mikedn commented:
Hmm, looks like runtest.proj depends on the .nugget directory somehow: https://github.com/dotnet/coreclr/blob/master/tests/runtest.proj#L362. For what it's not clear, tests run fine. Do we even need that .json file anymore?

mikedn commented:
tests.zip produced by a Windows debug build is 300 megabytes smaller:

https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_windows_nt_prtest/7518/artifact/bin/tests/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_windows_nt_prtest/7519/artifact/bin/tests/

BruceForstall commented:
@wtgodbe PTAL

wtgodbe commented:
I'm 99% sure it won't, but there's a chance that the against-packages test build will be broken by this (that scenario isn't tested in CI). I'm running that build locally now, assuming that succeeds, LGTM.

wtgodbe commented:
Actually, never mind, I thought that your change was in build-test, not runtest. You may want to make a similar change here - https://github.com/dotnet/coreclr/blob/master/build-test.cmd#L289

wtgodbe commented:
Local build succeeded with your change & the change I mentioned above

mikedn commented:
So if it works does that mean that the json file generate by runtest.proj is no longer needed?

wtgodbe commented:
We don't even restore project.json's anymore (we use .csproj's instead), so we should be able to get rid of that generated project.json.

mikedn commented:
I've updated build-test.cmd and removed the JSON building target. 

As far as I can tell the OSX build failure is not related to these changes, just some Jenkins issue.

mikedn commented:
@wtgodbe Good to go?

wtgodbe commented:
Looks fine to me, pending:

@dotnet-bot test OSX10.12 x64 Checked Build and Test
